packages/shared/src/hooks/streaks/useStreakMilestone.ts

```diff
@@ -42,5 +45,11 @@ export const useStreakMilestone = (): void => {
         },
       },
     });
-  }, [modalType, openModal, shouldHideModal, streak, updateAlerts]);
+
+    trackEvent({
```

Comment:
When is this supposed to be tracked. I see this in analytics but doesn't make much sense to me `When the shortcut component renders`. Also left a comment for Dave there

Reply:
Probably, when the streaks visual element renders.
But it's pretty much the same as the click then? 👀

Reply:
I was thinking when the button renders, then it wouldn't be the same as we click one. But not sure, maybe it's for any other modal from streaks showing. Let's confirm with Dave 🙏

Reply:
Yeah after looking I think it's the impression for the modal.
As he handles the clicks/dismissal seperate.

Reply:
Makes sense from the descriptions as well. Just here concerned about what I mentioned above then since this is in a useEffect.

Reply:
@nensidosari I'm not sure we have an alternative to a side effect here, since these modals are displayed on layout mount and not on a click handler -- see this: https://github.com/dailydotdev/apps/blob/083c4ce047accfd19645978bc0ffc4b47b941bd1/packages/shared/src/components/MainLayout.tsx#L91

An alternative would be to track the event on the `FirstStreakModal` and `NewStreakModal` component, I chose to do it here to avoid duplicating code.

Reply:
You can probably add a ref just to see if it already fired? 🙏

See [here](https://github.com/dailydotdev/apps/commit/cb61c985acf3a93b2ef9877eb3f65ec4e3320ccb)